File: ServerFacade/src/main/java/Result/UserResult.java
package Result;

import Model.User;

public class UserResult extends Result {
  private User user;

  public UserResult(boolean success, String message) {
    super(success, message);
  }

  public UserResult(User user) {
    super(true, "success");
    this.user = user;
  }

  public User getUser() {
    return user;
  }
}

File: FrontEnd/lib/Result/UserResult.dart
import 'package:flutter_app/Model/User.dart';
import 'package:json_annotation/json_annotation.dart';

import 'Result.dart';

part 'UserResult.g.dart';

@JsonSerializable(explicitToJson: true)

class UserResult extends Result {
  User user;

  UserResult(bool success, String message) : super(success, message);

  UserResult.user(this.user) : super(true, "success");

  factory UserResult.fromJson(Map<String, dynamic> json) => _$UserResultFromJson(json);

  Map<String, dynamic> toJson() => _$UserResultToJson(this);
}